Classes/EnricherRegistry.php

```diff
@@ -14,36 +14,30 @@ class EnricherRegistry implements EnricherInterface
 {
     private static $objects = [];
 
-    public static function register($class, $priority = 0)
+    public static function register($className, $priority = 0)
     {
-        if (!is_a($class, EnricherInterface::class, true)) {
-            throw new \LogicException(sprintf('The class "%s" does not implements "%s".', $class, EnricherInterface::class));
+        if (!is_a($className, EnricherInterface::class, true)) {
+            throw new \LogicException(sprintf('The class "%s" does not implements "%s".', $className, EnricherInterface::class));
         }
 
         if (!\array_key_exists($priority, self::$objects)) {
             self::$objects[$priority] = [];
         }
 
-        self::$objects[$priority][] = $class;
+        self::$objects[$priority][$className] = true;
     }
 
-    public static function unregister($class, $priority = 0)
+    public static function unregister($className, $priority = 0)
     {
-        if (!isset(self::$objects[$priority])) {
-            return;
+        if (self::$objects[$priority][$className] ?? false) {
+            unset(self::$objects[$priority][$className]);
         }
-
-        if (!isset(self::$objects[$priority][$class])) {
-            return;
-        }
-
-        unset(self::$objects[$priority][$class]);
     }
 
     public function process(FrontendUser $user, array $context)
     {
-        foreach ($this->flatten(self::$objects) as $className) {
-            GeneralUtility::makeInstance($className)->process($user, $context);
+        foreach ($this->flatten(self::$objects) as $className => $value) {
```

Comment:
To avoid such an unused variable like $value I would suggest to put the class name in line 27 into the array.

Reply:
@hannesbochmann I made the change. Do you have any further suggestions or can we merge this?